src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.StateKind.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    internal partial class AnalysisState
+    {
+        /// <summary>
+        /// State kind of per-analyzer <see cref="AnalyzerStateData"/> tracking an analyzer's partial analysis state.
+        /// </summary>
+        internal enum StateKind
+        {
+            /// <summary>
+            /// Ready for processing.
```

Comment:
@JohnHamby I agree with your comments, this needs renaming and better comments.
@heejaechang Null state means not yet processed at all. A non-null value means either it is currently being processed or partially processed (either due to cancellation, or partial analysis of symbol declarations) and ready for someone to process.